src/reference/asciidoc/amqp.adoc

```diff
@@ -361,6 +361,19 @@ The `ConnectionFactory` argument can be used to distinguish target connection na
 By default a `beanName` of the `AbstractConnectionFactory` and an internal counter are used to generate `connection_name`.
 The `<rabbit:connection-factory>` namespace component is also supplied with the `connection-name-strategy` attribute.
 
+
+The connection might be blocked for interaction from the Broker according to the https://www.rabbitmq.com/memory.html[Memory Alarm].
+Starting with _version 2.0_, the `org.springframework.amqp.rabbit.connection.Connection` can be supplied with the `com.rabbitmq.client.BlockedListener` s to to be notified for connection block and unblock events.
+In addition the `AbstractConnectionFactory` emits a `ConnectionBlockedEvent` and `ConnectionUnblockedEvent`, respectively, via its internal `BlockedListener` implementation.
+These can be useful in application logic to react appropriately to the problems on the Broker and perform some maintenance actions for example.
+
+IMPORTANT: When single `CachingConnectionFactory` is used in the application, e.g. in case of auto-configuration by Spring Boot, it is important to understand that by default it provides a single shared connection instance.
+And when it is blocked by the Broker, any its clients stop to work.
+If we deal with producers and consumers in the same application, we may end up with the dead lock situation when producers are blocking connection because no resources on Broker any more and consumers can't free them because connection is blocked.
+To mitigate the problem, there is just enough to have one more separate `CachingConnectionFactory` instance with the same options - one for producers and one for consumers.
+At the same time this isn't recommended if we use transactional producer and it is important to let that producer to reuse a `Channel` associated with the consumer transaction.
+
```

Comment:
This reads better, but I'm not sure what you mean:  What isn't recommended?
```At the same time this isn't recommended when we use a transactional producer and it is important to let the producer reuse a `Channel` associated with the consumer transaction.```

Reply:
Good. How about this?

> The separate `CachingConnectionFactory` isn't recommended for transactional producers, since they should reuse a `Channel` associated with the consumer transaction.